tests/unit/test_configuration.py

```diff
@@ -42,8 +49,73 @@ def test_get_configuration_with_user_override(user_configuration_yaml, mocker):
     update_calls = [
         call
         for call in mock.mock_calls
-        if "update" in str(call)
-        and "user" in str(call)
-        and str(user_configuration_yaml) in str(call)
```

Comment:
The `get_configuration` function is fundamentally changed since it no longer calls .update() on a path. I strengthened this assertion up as much as I could given that we lost the user_configuration_yaml object.